prefect_sqlalchemy/credentials.py

```diff
@@ -225,31 +345,507 @@ def sqlalchemy_credentials_flow():
+                where each row is a tuple and each column is a value in the tuple.
+
+        Examples:
+            Create a table, insert three rows into it, and fetch all where name is 'Me'.
+            ```python
+            from prefect_sqlalchemy import DatabaseCredentials
+
+            with DatabaseCredentials.load("MY_BLOCK") as database:
+                database.execute("CREATE TABLE IF NOT EXISTS customers (name varchar, address varchar);")
+                database.execute_many(
+                    "INSERT INTO customers (name, address) VALUES (:name, :address);",
+                    seq_of_parameters=[
+                        {"name": "Ford", "address": "Highway 42"},
+                        {"name": "Unknown", "address": "Space"},
+                        {"name": "Me", "address": "Myway 88"},
+                    ],
+                )
+                results = database.fetch_all("SELECT * FROM customers WHERE name = :name", parameters={"name": "Me"})
+            ```
+        """  # noqa
+        result_set = await self._get_result_set(
+            text(operation), parameters, execution_options=execution_options
+        )
+        rows = result_set.fetchall()
+        return rows
+
+    @sync_compatible
+    async def execute(
+        self,
+        operation: str,
+        parameters: Optional[Dict[str, Any]] = None,
+        **execution_options: Dict[str, Any],
+    ) -> None:
+        """
+        Executes an operation on the database. This method is intended to be used
+        for operations that do not return data, such as INSERT, UPDATE, or DELETE.
+
+        Unlike the fetch methods, this method will always execute the operation
+        upon calling.
+
+        Args:
+            operation: The SQL query or other operation to be executed.
+            parameters: The parameters for the operation.
+            **execution_options: Options to pass to `Connection.execution_options`.
+
+        Examples:
+            Create a table and insert one row into it.
+            ```python
+            from prefect_sqlalchemy import DatabaseCredentials
+
+            with DatabaseCredentials.load("MY_BLOCK") as database:
+                database.execute("CREATE TABLE IF NOT EXISTS customers (name varchar, address varchar);")
+                database.execute(
+                    "INSERT INTO customers (name, address) VALUES (:name, :address);",
+                    parameters={"name": "Marvin", "address": "Highway 42"},
+                )
+            ```
+        """  # noqa
+        async with self._manage_connection(begin=False) as connection:
+            await self._async_sync_execute(
+                connection,
+                text(operation),
+                parameters,
+                execution_options=execution_options,
+            )
+
+    @sync_compatible
+    async def execute_many(
+        self,
+        operation: str,
+        seq_of_parameters: Optional[List[Dict[str, Any]]],
```

Comment:
this feels like a slightly unclear name, not exactly sure what I'd use instead, maybe `list_of_parameters` since it's actually a list? not sure